src/gpu/intel/jit/conv/config.cpp

```diff
@@ -599,6 +600,26 @@ bool is_plain_tag_optimal_for_output(
     return false;
 }
 
+bool is_optimal_abx_load(
+        tensor_kind_t kind, ngen::HW hw, const conv_problem_t &prb) {
+    gpu_assert(utils::one_of(kind, tensor_kind_t::src, tensor_kind_t::dst));
+    if (hw < ngen::HW::XeHPC) return false;
+    bool is_src = (kind == tensor_kind_t::src);
+    if (is_src) {
+        if (prb.kw != 1 || prb.sw != 1) return false;
+    }
+    dim_t c = prb.g * (is_src ? prb.ic : prb.oc);
+    dim_t d = (is_src ? prb.id : prb.od);
+    dim_t h = (is_src ? prb.ih : prb.oh);
+    dim_t w = (is_src ? prb.iw : prb.ow);
+    int type_size = into<int>(types::data_type_size(
+            is_src ? prb.src_data_type : prb.dst_data_type));
+    if (!block_2d_width_ok(w, type_size)) return false;
+    if (!block_2d_height_ok(c)) return false;
+    if (!block_2d_pitch_ok(hw, d * h * w, type_size)) return false;
```

Comment:
Rather than basing this on tags, would it make sense to just rely on the strides from the memory descriptor? If we did that, then this check would also apply to any kind of strided layouts as well.

Reply:
In the current implementation any strided layouts are handled in a very simple way: they are forwarded to the kernel as is:
https://github.com/uxlfoundation/oneDNN/blob/52c9e61803ac89269488bb493258b109276a3a6c/src/gpu/intel/jit/conv/config.cpp#L676

So this should be already there: if any block 2D messages can be used - they are used. It's not optimal in general (e.g. in some cases pre-reorder would give better performance), but it seems good enough for now, at least to have functional support. We haven't been getting performance requests on strided layout performance so looks like good enough for now.

Reply:
I agree, in general we better rely on W-dense or C-dense layouts instead of NCHW/NHWC - this generalization makes perfect sense. But the changes in this PR are quite simple though.
I suggest to get back to dense/strided layout generalization when we get more requests about strided layouts support. I feel like we don't get enough requests on that to justify more work.